Content.Server/Physics/Controllers/ConveyorController.cs

```diff
@@ -1,51 +1,161 @@
+            }
+        }
+
+        private void OnConveyorStartCollide(EntityUid uid, ConveyorComponent component, StartCollideEvent args)
+        {
+            var otherUid = args.OtherFixture.Body.Owner;
+
+            if (args.OtherFixture.Body.BodyType == BodyType.Static) return;
+
+            component.Intersecting.Add(otherUid);
+            EnsureComp<ActiveConveyorComponent>(uid);
+        }
+
+        private void OnInit(EntityUid uid, ConveyorComponent component, ComponentInit args)
+        {
+            _signalSystem.EnsureReceiverPorts(uid, component.ReversePort, component.ForwardPort, component.OffPort);
+
+            if (TryComp<PhysicsComponent>(uid, out var body))
+            {
+                var shape = new PolygonShape();
+                shape.SetAsBox(0.55f, 0.55f);
+
+                _fixtures.TryCreateFixture(body, new Fixture(body, shape)
+                {
+                    ID = ConveyorFixture,
+                    CollisionLayer = (int) (CollisionGroup.LowImpassable | CollisionGroup.MidImpassable | CollisionGroup.Impassable),
+                    Hard = false,
+                });
+            }
+        }
+
+        private void OnPowerChanged(EntityUid uid, ConveyorComponent component, PowerChangedEvent args)
+        {
+            UpdateAppearance(component);
+        }
+
+        private void UpdateAppearance(ConveyorComponent component)
+        {
+            if (!EntityManager.TryGetComponent<AppearanceComponent?>(component.Owner, out var appearance)) return;
+            var isPowered = this.IsPowered(component.Owner, EntityManager);
+            appearance.SetData(ConveyorVisuals.State, isPowered ? component.State : ConveyorState.Off);
+        }
+
+        private void OnSignalReceived(EntityUid uid, ConveyorComponent component, SignalReceivedEvent args)
+        {
+            if (args.Port == component.OffPort)
+                SetState(component, ConveyorState.Off);
+            else if (args.Port == component.ForwardPort)
+                SetState(component, ConveyorState.Forward);
+            else if (args.Port == component.ReversePort)
+                SetState(component, ConveyorState.Reverse);
+        }
+
+        private void SetState(ConveyorComponent component, ConveyorState state)
+        {
+            component.State = state;
+
+            if (TryComp<RecyclerComponent>(component.Owner, out var recycler))
+            {
+                if (component.State != ConveyorState.Off)
+                    _recycler.EnableRecycler(recycler);
+                else
+                    _recycler.DisableRecycler(recycler);
+            }
+
+            UpdateAppearance(component);
+        }
+
+        public bool CanRun(ConveyorComponent component)
+        {
+            return component.State != ConveyorState.Off &&
+                   !EntityManager.HasComponent<SharedItemComponent>(component.Owner) &&
+                   this.IsPowered(component.Owner, EntityManager);
+        }
```

Comment:
Whats with the item component check? If conveyor items cause issues, there's probably a better way of stopping that than having to check each tick if a conveyor is an item?

If this is really needed, maybe just set the state to off when an item comp gets added & preventing it from being enabled while it stays an item?

Reply:
I think this was there previously in case it got picked up.